asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -1318,27 +1283,6 @@ machine Main {
     foo;
   }
 }
-";
-        parse_analyse_and_compile::<GoldilocksField>(asm);
-    }
-
-    #[test]
-    #[should_panic(expected = "rhs of external instruction can't have repeated output registers")]
-    fn instr_external_rhs_repeated_output_register() {
-        let asm = r"
-machine Main {
-  degree 8;
-  reg pc[@pc];
-  reg X[<=];
-  reg Y[<=];
-  reg A;
-
-  instr foo X  = vm.foo X -> A, A;
```

Comment:
Doesn't this still make sense?

Reply:
I don't think it does, because now everything can be a complex expression, with multiple registers, etc.
This is kind of I was hinting at in chat:
With expressions we give total freedom to the user to create the lookup it wants, so it probably doesn't make sense to try and figure out these corner cases (e.g., maybe the user does want both outputs to be equal)